constraint/cmd/cfctc/cfctc.go

```diff
@@ -0,0 +1,130 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+
+	"github.com/golang/glog"
+	_ "github.com/golang/glog"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+
+	"github.com/open-policy-agent/frameworks/constraint/pkg/compiler/regorewriter"
+)
+
+var rootCmd = &cobra.Command{
+	Use:   "cfctc",
+	Short: "cfctc is the constraint framework constraint template compiler",
+	Long: `
+Example usage for transforming the forseti-security/policy-library constraints:
+git clone git@github.com:forseti-security/policy-library.git
+go run ./cmd/cfctc/cfctc.go \
+  --ct policy-library/validator \
+  --lib policy-library/lib \
+  --input ./policy-library \
+  --output ./policy-library-rewrite \
+  --pkgPrefix x.y.z \
+  --alsologtostderr
+opa test -v rewrite/lib/ rewrite/validator/
+meld policy-library/lib/ rewrite/lib/
+meld policy-library/validator/ rewrite/validator/
+`,
+	Run: rootCmdFn,
+}
+
+var (
+	cts       []string
+	libs      []string
+	pkgPrefix string
+	oldRoot   string
+	newRoot   string
+)
+
+func init() {
+	rootCmd.Flags().StringSliceVar(
+		&cts, "ct", nil, "The rego for the constraint template body")
```

Comment:
@maxsmythe can you clarify what you mean by "side-load"?

Reply:
Inject the Rego code to be executed alongside the query to be executed.

Reply:
Hrmmm. I'm not sure I understand then. As far as the flag goes, it would be nice if we didn't have to add another for mutation templates. If the goal is to just separate "template" code from "library" code let's rename just rename the parameter. We can have the mutation discussion elsewhere (I don't think it would affect this?)

Reply:
Agreed this is a side convo